### DISCUSSION TITLE
Integrating Perplexity search

### DISCUSSION THREAD

**Discussion by Hi-Angel** [qutebrowser/qutebrowser#8435] [General] [answered]
In my QB, I have simultaneously 7 search engines for different purposes, which is very useful.

[Perplexity](https://www.perplexity.ai) is an AI-based search engine. Turns out to be useful sometimes, so I wanted to add it as another one in my QB. Found so far no way to do that though.

It seems Perplexity doesn't accept "query parameters" in URL, and looking at network communication via <kbd>F12</kbd> I don't seem to see anything easily automatable.

Might be not be easy, but I figured I'd at least start the discussion, so if anyone finds solution, it could be shared with everyone 😊

**Comment by jnphilipp** [comment, marked as answer]
I have Perplexity set as a search engine with the following URL: `https://www.perplexity.ai/search/?q={}`, works great.

**Reply by Hi-Angel** [reply]
Oh, thank you, indeed that works! I was trying a `https://www.perplexity.ai/search/text-follows` *(as it's been a URL I've seen)*. So it turns a `?q=` has to be used. I haven't seen such way, thank you for pointing that out!

**Comment by Hi-Angel** [comment]
To add to the answer: to enable pro-mode add ~~`&copilot=true`, i.e.: `https://www.perplexity.ai/search/?q={}&copilot=true`~~

**UPD**: as of 1 Feb 2025 `copilot` no longer does anything. I reported it to them, waiting for the answer.

**Reply by kkotsche1** [reply]
Have you heard anything here? Currently trying to figure out if there is still some way to enable deep research within the URL

**Reply by Hi-Angel** [reply]
Yeah, they did fix it a month after I posted this comment, so it worked.

Now that I'm trying though, it doesn't seem to work — the "pro queries remaining" isn't get exhausted even though I do a pro query. Apparently they broke it again.
